content/academy/apify_platform/deploying/docker_file.md

```diff
@@ -0,0 +1,119 @@
+---
+title: Dockerfile
+description: Understand how to write a Dockerfile (Docker image blueprint) for your project so that it can be run within a Docker container on the Apify platform.
+menuWeight: 3
+paths:
+- apify-platform/deploying/docker-file
+---
+
+# [](#dockerfile) Dockerfile
+
+The **Dockerfile** is a file which gives the Apify platform (or Docker, more specifically) instructions on how to create an environment for your code to run in. Every actor must have a Dockerfile, as actors run in Docker containers.
+
+> Actors on the platform are always run in Docker containers; however, they can also be run in local Docker containers. This is not common pracice though, as it requires more setup and a deeper understanding of Docker. For testing, it's best to just run the actor on the local OS (this requires you to have the underlying runtime installed, such as Node.js, Python, Rust, GO, etc).
+
+## [](#base-images) Base images
+
+If your project doesn’t already contain a Dockerfile, don’t worry! [Docker Hub](https://hub.docker.com/) provides a ton of free Docker images for most use-cases, upon which you can create your own images. At the base level, each Docker image contains a Linux OS distribution and usually also a programming language runtime (such as Node.js). You can also find images with preinstalled libraries or just install them yourself during the build step.
+
+> Apify offers many base images for building actors, which can be found [here](https://hub.docker.com/u/apify)
```

Comment:
I would flip the sequence here. I would first say that "Apify offers base images that are optimized for running on the Apify platform bla bla..." And only after that talk about the Docker Hub. In reality, if the devs use a different image than one of our own, the actor startup times will be horrible, so we need to strongly nudge them to use our own images and only use Docker Hub as a backup. I would go even further and tell them to try to use our images (because they're optimized) even if they already have their own Dockerfiles.